osu.Game.Rulesets.Taiko/Objects/DrumRoll.cs

```diff
@@ -86,7 +86,7 @@ private void createTicks(CancellationToken cancellationToken)
 
             bool first = true;
 
-            for (double t = StartTime; t < EndTime + tickSpacing / 2; t += tickSpacing)
+            for (double t = StartTime; t < EndTime + tickSpacing; t += tickSpacing)
```

Comment:
The `TickRate` change I get, but what's happened here with the termination condition of the for loop? I see in history you had `Precision.AlmostEquals()` there first, but now this? I would think that would make the issue potentially *worse* and not better...?

Reply:
The `AlmostEquals` was a fix before I understood how lazer worked. I didn't immediately find the `TickRate` application in the `TaikoBeatmapConverter` until smoogi informed me of it, so I was under the assumption that we were just doing a best-effort.

The change from `tickSpacing / 2` to `tickSpacing` is to match the stable implementation (reference: https://github.com/peppy/osu-stable-reference/blob/1531237b63392e82c003c712faa028406073aa8f/osu!/GameplayElements/HitObjects/Taiko/SliderTaiko.cs#L284), and in theory should be closer to intended behaviour than what we had.

You would think this would make things worse, but alongside the actual fix (making `TickRate` update on newly placed drum rolls in the editor) it doesn't seem to have any adverse effects.

I'm okay with reverting to `tickSpacing/2` if it feels more defensively correct to you. I don't think this will make a huge difference in the 99.99% of cases.

Reply:
I've applied the revert myself in https://github.com/ppy/osu/pull/23768/commits/7acd186a3d78a6e2ae99446eaf8d8dd89c2dbb8b.